core/application/src/query_runner.rs

```diff
@@ -82,198 +82,102 @@ impl QueryRunner {
-        &self,
-        public_key: &NodePublicKey,
-        f: F,
-    ) -> Option<T> {
-        self.inner.run(|ctx| {
-            self.pub_key_to_index
-                .get(ctx)
-                .get(public_key)
-                .map(|index| self.node_table.get(ctx).get(index).map(f))
-                .unwrap_or(None)
-        })
-    }
 }
 
 impl SyncQueryRunnerInterface for QueryRunner {
-    fn get_account_balance(&self, account: &EthAddress) -> u128 {
-        self.inner.run(|ctx| {
-            self.account_table
-                .get(ctx)
-                .get(account)
-                .map(|a| a.bandwidth_balance)
-                .unwrap_or(0)
-        })
-    }
-    fn get_client_balance(&self, client: &ClientPublicKey) -> u128 {
-        self.inner.run(|ctx| {
-            let client_table = self.client_table.get(ctx);
-            let account_table = self.account_table.get(ctx);
-            // Lookup the account key in the client->account table and then check the balance on the
-            // account
-            client_table
-                .get(client)
-                .and_then(|key| account_table.get(key))
-                .map(|a| a.bandwidth_balance)
-                .unwrap_or(0)
-        })
+    fn get_metadata(&self, key: &Metadata) -> Option<Value> {
+        self.inner.run(|ctx| self.metadata_table.get(ctx).get(key))
     }
 
-    fn get_flk_balance(&self, account: &EthAddress) -> HpUfixed<18> {
-        self.inner.run(|ctx| {
-            self.account_table
-                .get(ctx)
-                .get(account)
-                .map(|account| account.flk_balance)
-                .unwrap_or(HpUfixed::<18>::zero())
-        })
-    }
-
-    fn get_stables_balance(&self, account: &EthAddress) -> HpUfixed<6> {
-        self.inner.run(|ctx| {
-            self.account_table
-                .get(ctx)
-                .get(account)
-                .map(|account| account.stables_balance)
-                .unwrap_or(HpUfixed::<6>::zero())
-        })
-    }
-
-    fn get_staked(&self, node: &NodePublicKey) -> HpUfixed<18> {
-        self.get_node_info_with_pub_key(node, |node_info| node_info.stake.staked)
-            .unwrap_or(HpUfixed::zero())
-    }
-
-    fn get_locked(&self, node: &NodePublicKey) -> HpUfixed<18> {
-        self.get_node_info_with_pub_key(node, |node_info| node_info.stake.locked)
-            .unwrap_or(HpUfixed::zero())
+    fn get_account_info<F: Clone>(
+        &self,
+        address: &EthAddress,
+        selector: impl FnOnce(AccountInfo) -> F,
+    ) -> Option<F> {
+        self.inner
+            .run(|ctx| self.account_table.get(ctx).get(address))
+            .map(selector)
     }
 
-    fn get_stake_locked_until(&self, node: &NodePublicKey) -> Epoch {
-        self.get_node_info_with_pub_key(node, |node_info| node_info.stake.stake_locked_until)
-            .unwrap_or(0)
+    fn get_client_info(self, pub_key: &ClientPublicKey) -> Option<EthAddress> {
```

Comment:
Nit: Since this table is mapping ClientKey->EthAddress(AccountAddress) lets rename this function in interface to client_key_to_account_key